src/amenities/amenities_server/controller/AmenitiesController.cs

```diff
@@ -0,0 +1,104 @@
+using Microsoft.AspNetCore.Mvc;
+
+namespace amenities_server.Controllers;
+
+[ApiController]
+[Route("amenities")]
+public class AmenitiesController : ControllerBase
+{
+    private IAmenityPersistence _amenityPersistence;
+
+    public AmenitiesController()
+    {
+        _amenityPersistence = Services.GetAmenityPersistence();
+    }
+
+    //get: /amenities
+    [HttpGet]
+    public IActionResult GetAmenities()
+    {
+        var amenities = _amenityPersistence.GetAmenities();
+
+        if (amenities == null)
+        {
+            return NotFound();
+        }
+
+        return Ok(new AmenityResponse<IEnumerable<Amenity>>("Amenities retrieved successfully.", amenities));
```

Comment:
Consider extracting all these response strings into a static class to help with consistency